.github/workflows/docker-publish.yml

```diff
@@ -0,0 +1,112 @@
+name: Docker
+
+# This workflow uses actions that are not certified by GitHub.
+# They are provided by a third-party and are governed by
+# separate terms of service, privacy policy, and support
+# documentation.
+
+on:
+  push:
+    branches:
+      - main
+    # Publish semver tags as releases.
+    tags:
+      - v*.*.*
+    paths:
+      - datadog/apps/images/**
+  pull_request:
+    branches:
+      - main
+    paths:
+      - datadog/apps/images/**
+
+env:
+  # Use docker.io for Docker Hub if empty
+  REGISTRY: ghcr.io
+  # github.repository as <account>/<repo>
+  IMAGE_NAMESPACE: ${{ github.repository_owner }}
+
+
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
+    permissions:
+      contents: read
+      packages: write
+      # This is used to complete the identity challenge
+      # with sigstore/fulcio when running outside of PRs.
+      id-token: write
+
+    strategy:
+      fail-fast: false
+      matrix:
+        apps:
+          - dogstatsd
+          - http-client
+          - nginx-server
+          - prometheus
+          - redis-client
```

Comment:
What about building these in https://github.com/DataDog/docker-library instead ? Do you except these to change often ?

Reply:
The content of the docker image is as important as the manifest (pulumi code to create the kubernetes pod or ECS task) in order to determine what the application will do and what the agent will do with that workload. And both are often strongly linked.

For example, an application that exposes a Prometheus endpoint needs to set the port in the app source code and in the k8s manifest.
The exact same port value needs to be set here: https://github.com/DataDog/test-infra-definitions/pull/99/files#diff-72c7db7015a1d7c8eecc2486e0bbb760f1fa19cd2fb1f0ab1c4182505efd6672R18 and here: https://github.com/DataDog/test-infra-definitions/pull/99/files#diff-2bf5b21c2f389ead885e1ef5b42a328316e7385401da6e38c559201c72a47eacR81
That’s why I’d rather keep the two close of each other.

Similarly, for an application that emits dogstatsd data, the application needs to know in its source code where is the dogstatsd endpoint and it is set thanks to the k8s downward API in the manifest.
The exact same environment variable name needs to be used here: https://github.com/DataDog/test-infra-definitions/pull/99/files#diff-013ee6d71017bc4436331edbb8d927d71b3ec92f9c1b2968da5f2b341458051fR25 and here https://github.com/DataDog/test-infra-definitions/pull/99/files#diff-a51f5b99d60fcbbd5161bd40b083d24e97da6416dce1f8b0eea3ffd90217dd35R67
Here again, I’d rather keep the two close of each other.

Finally, I’d like to avoid today’s situation where we have several repositories containing some kubernetes manifests that are referencing docker images for which we just have lost the source because they were pushed manually once.

This `docker-library` repository doesn’t seem to have any CI to build and automatically push images.